File: c/hash-table/count-primes.c
int countPrimes(int n) {
    int *arr;
    int count = 0;
    int i,j;

    arr = malloc(sizeof(int) * n);
    memset(arr, 0, sizeof(int) * n);
    for(i = 2; i < n; ++i){
        if(arr[i])
            continue;
        ++count;
        for(j = i * 2; j < n; j += i)
            arr[j] = 1;
    }

    free(arr);
    return count;
}

File: python/hash-table/count-primes.py
class Solution(object):
    def countPrimes(self, n):
        arr = [0] * n
        count = 0
        for i in xrange(2,n):
            if arr[i] != 0:
                continue
            count += 1
            for j in xrange(2*i,n,i):
                arr[j] = 1

        return count
